neqo-http3/src/connection.rs

```diff
@@ -630,8 +630,9 @@ impl Http3Connection {
             .http_stream()
             .ok_or(Error::InvalidStreamId)?;
 
-        if stream.priority() != priority {
-            stream.priority_update(priority);
+        let priority_handler = stream.priority_handler_mut();
+        if priority_handler.priority() != priority {
+            priority_handler.priority_update(priority);
```

Comment:
This can all be in a single function maybe_update_priority that returns true if priority is updated otherwise false.